framework/test/unit/modules/interoperability/sidechain/module.spec.ts

```diff
@@ -0,0 +1,587 @@
+import { codec } from '@liskhq/lisk-codec';
+import { BlockAssets } from '@liskhq/lisk-chain';
+import { utils } from '@liskhq/lisk-cryptography';
+import { GenesisInteroperability } from '../../../../../src/modules/interoperability/types';
+import {
+	CreateGenesisBlockContextParams,
+	createGenesisBlockContext,
+	InMemoryPrefixedStateDB,
+} from '../../../../../src/testing';
+import {
+	GenesisBlockExecuteContext,
+	MODULE_NAME_INTEROPERABILITY,
+	SidechainInteroperabilityModule,
+	ChainStatus,
+} from '../../../../../src';
+import { genesisInteroperabilitySchema } from '../../../../../src/modules/interoperability/schemas';
+import {
+	genesisInteroperability,
+	terminatedStateAccount,
+	terminatedOutboxAccount,
+	chainInfo,
+	chainData,
+	lastCertificate,
+	chainValidators,
+	activeValidator,
+} from '../interopFixtures';
+import { PrefixedStateReadWriter } from '../../../../../src/state_machine/prefixed_state_read_writer';
+import {
+	getMainchainID,
+	validNameCharset,
+	getMainchainTokenID,
+	computeValidatorsHash,
+} from '../../../../../src/modules/interoperability/utils';
+import {
+	MIN_CHAIN_NAME_LENGTH,
+	MAX_CHAIN_NAME_LENGTH,
+	CHAIN_NAME_MAINCHAIN,
+	EMPTY_HASH,
+	HASH_LENGTH,
+	CHAIN_ID_LENGTH,
+} from '../../../../../src/modules/interoperability/constants';
+import { OwnChainAccountStore } from '../../../../../src/modules/interoperability/stores/own_chain_account';
+
+const createInitGenesisStateContext = (
+	genesisInterop: GenesisInteroperability,
+	params: CreateGenesisBlockContextParams,
+): GenesisBlockExecuteContext => {
+	const encodedAsset = codec.encode(genesisInteroperabilitySchema, genesisInterop);
+
+	return createGenesisBlockContext({
+		...params,
+		assets: new BlockAssets([{ module: MODULE_NAME_INTEROPERABILITY, data: encodedAsset }]),
+	}).createInitGenesisStateContext();
+};
+
+describe('initGenesisState', () => {
+	const chainID = Buffer.from([1, 2, 3, 4]);
+
+	let params: CreateGenesisBlockContextParams;
+	let stateStore: PrefixedStateReadWriter;
+	let interopMod: SidechainInteroperabilityModule;
+
+	const ownChainAccountStoreMock = {
+		get: jest.fn(),
+		set: jest.fn(),
+		has: jest.fn(),
+	};
+
+	beforeEach(() => {
+		stateStore = new PrefixedStateReadWriter(new InMemoryPrefixedStateDB());
+		interopMod = new SidechainInteroperabilityModule();
+		params = {
+			stateStore,
+			chainID,
+		};
+
+		interopMod.stores.register(OwnChainAccountStore, ownChainAccountStoreMock as never);
+	});
+
+	describe('If chainInfos is empty', () => {
```

Comment:
I think we have the pattern `describe(when ...)` in the code elsewhere, also in the other places.
```suggestion
	describe('when chainInfos is empty', () => {
```